qiskit_experiments/calibration_management/basis_gate_library.py

```diff
@@ -289,3 +321,117 @@ def default_values(self) -> List[DefaultCalValue]:
                     defaults.append(DefaultCalValue(value, param.name, tuple(), name))
 
         return defaults
+
+
+class EchoedCrossResonance(BasisGateLibrary):
+    """A fixed-frequency transmon gate library with echoed cross-resonance gates.
+
+    This library extends the FixedFrequencyTransmon library by adding support for
+    cross-resonance gates.
+    """
+
+    __default_values__ = {
+        "cr_duration": 640,
+        "cr_width": 384,
+        "cr_σ": 64,
+        "cr_amp": 0.5,
+        "rot_amp": 0.0,
+    }
+
+    def __init__(
+        self,
+        single_qubit_library: BasisGateLibrary,
+        pulse_on_target: bool = True,
+        basis_gates: Optional[List[str]] = None,
+        default_values: Optional[Dict] = None,
+    ):
+        """Setup the schedules.
+
+        Args:
+            single_qubit_library: The library with the single-qubit gates from which
+                the X-gate schedule will be called to build the echo. This library needs
+                to supply a single-qubit schedule named ``"x"``.
+            pulse_on_target: A boolean to indicate if a rotary/cancellation tones are
+                used in the CR gate. If this value is set to False then no pulse is played
+                on the target qubit during the cross-resonance pulse.
+            basis_gates: The basis gates to generate.
+            default_values: Default values for the parameters this dictionary can contain
+                the following keys: "duration", "amp", "β", and "σ". If "σ" is not provided
+                this library will take one fourth of the pulse duration as default value.
+        """
+        self._rotary = pulse_on_target
+        super().__init__(basis_gates, default_values, [single_qubit_library])
+
+    @property
+    def __supported_gates__(self) -> Dict[str, int]:
+        """The gates that this library supports."""
+        return {"ecr": 2, "cr90p": 2, "cr90m": 2}
+
+    def _build_schedules(
+        self,
+        basis_gates: Set[str],
+        other_libraries: Optional[List["BasisGateLibrary"]] = None,
+    ) -> Dict[str, ScheduleBlock]:
+        """Build the schedules of the library."""
+        if "x" not in other_libraries[0]:
+            raise CalibrationError("x gate is required to build cross-resonance schedules.")
+
+        target = pulse.DriveChannel(Parameter("ch1"))
+        cr_chan = pulse.ControlChannel(Parameter("ch0.1"))
+
+        cr_amp = Parameter("cr_amp")
+        rot_amp = Parameter("rot_amp")
+        cr_sigma = Parameter("cr_σ")
+        cr_duration = Parameter("cr_duration")
+        cr_width = Parameter("cr_width")
+
+        cr90p = pulse.GaussianSquare(
```

Comment:
Since this is echoed cross-resonance, can we avoid the mistake of previous code and call this `cr45p`? Here the two pulses together make 90. This might help avoid confusion if we also add a library for unechoed cross-resonance where the single pulses really are a 90 degree rotation.

Reply:
Sounds good.